### PR TITLE
Add World Possible ScriptLauncher manifest

### PR DESCRIPTION
**Pull request by mwleeds**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

This is a copy of the upstream manifest
(https://github.com/endlessm/worldpossible-updutil/) with the branch
changed from main to the latest commit so the build is reproducible (or
at least that part of it is).

ScriptLauncher is an app we at Endless OS Foundation (endlessos.org)
created for World Possible (worldpossible.org) to assist them in their
administration of Endless OS computers that they have deployed in
various places (mostly for educational purposes in offline
environments). The function of it is to execute a Bash script as root
without requiring use of the Command Line Interface (though it can also
execute other things like Python scripts provided they have correct
permissions set and a correct shebang line). The reason World Possible
needs this is that they often have to make changes to deployed computers
such as installing or removing content, and the people available to push
the changes to each computer are not always experienced with the Linux
CLI.

Some consideration has been given to deploying changes with a technology
such as Ansible, but for now they are using shell scripts.

This utility requires an administrator password be entered to execute
the script as root, so it does not provide any privilege escalation.

It also has a few other features such as showing the output from the
script, but it is basically at MVP status.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by mwleeds** [comment]
Updated this to pick up https://github.com/endlessm/worldpossible-updutil/commit/2b0212f717e20c2f4bfcb9f7c4958e0614d6e1cc which I found by doing a test build.

File: org.worldpossible.ScriptLauncher.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  - "*.la"
+  - "*.a"
+
+modules:
+  - name: worldpossible-scriptlauncher
+    buildsystem: meson
+    builddir: true
+    sources:
+      - type: git
+        branch: 2b0212f717e20c2f4bfcb9f7c4958e0614d6e1cc
```
Comment (wjt) [inline org.worldpossible.ScriptLauncher.yml:34]: Does this work?

```suggestion
        commit: 2b0212f717e20c2f4bfcb9f7c4958e0614d6e1cc
```

Comment (mwleeds) [reply org.worldpossible.ScriptLauncher.yml:34]: I did it this way based on https://github.com/flathub/org.gnome.Builder/blob/master/org.gnome.Builder.json#L645

File: org.worldpossible.ScriptLauncher.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+app-id: org.worldpossible.ScriptLauncher
+runtime: org.gnome.Platform
+runtime-version: '3.38'
+branch: eos3
```
Comment (wjt) [inline org.worldpossible.ScriptLauncher.yml:4]: On Flathub all apps have branch `stable`. It's conventional to just not specify it:

```suggestion
```

**Review by wjt — CHANGES_REQUESTED** [review]

**Comment by wjt** [comment]
https://github.com/endlessm/worldpossible-scriptlauncher/blob/main/data/org.worldpossible.ScriptLauncher.appdata.xml.in will (I believe) need a `<releases>` section to pass the checker. Since we'll need to make up a version number, we should tag a version (1?) and then point the manifest at that tag, rather than at a commit sha, to satisfy https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds

File: org.worldpossible.ScriptLauncher.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
+  # We need to be able to run commands on the host to execute scripts chosen by the user
+  - --talk-name=org.freedesktop.Flatpak
+  - --filesystem=host
+
+build-options:
+  env: { V: '1' }
+
```
Comment (bilelmoussaoui) [inline org.worldpossible.ScriptLauncher.yml:20]: iirc, this is set by default by the sdk

Comment (mwleeds) [reply org.worldpossible.ScriptLauncher.yml]: Okay, removed it. It was just set by the Builder GNOME app template.

**Comment by LeandroStanger** [comment]
bot, build org.worldpossible.ScriptLauncher

**Comment by mwleeds** [comment]
I installed the test build and it inspired [one change](https://github.com/endlessm/worldpossible-scriptlauncher/commit/3172a3544564b7ac6a74b0f36c7d9eed289fe560) but otherwise seems good.

**Comment by mwleeds** [comment]
bot, build org.worldpossible.ScriptLauncher

**Comment by mwleeds** [comment]
Test build looks good to me

File: org.worldpossible.ScriptLauncher.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+app-id: org.worldpossible.ScriptLauncher
+runtime: org.gnome.Platform
+runtime-version: '3.38'
+sdk: org.gnome.Sdk
+command: worldpossible-scriptlauncher
+
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
```
Comment (barthalion) [inline org.worldpossible.ScriptLauncher.yml:9]:
```suggestion
```

Apparently this is part of `fallback-x11`.

**Comment by barthalion** [comment]
Who should be invited to the new repo?

**Comment by mwleeds** [comment]
> Who should be invited to the new repo?

I suppose myself and @wjt

File: org.worldpossible.ScriptLauncher.yml (hunk 4 above)
Comment (mwleeds) [reply org.worldpossible.ScriptLauncher.yml:9]: I don't see documentation of that in the man pages, or in the source code at a glance. Are you sure?

**Comment by wjt** [comment]
Is it possible to add a GitHub team from a different org as committers/owners of this repo? I fear the answer is "no, teams are private to an org". (But if yes, [this one](https://github.com/orgs/endlessm/teams/endless-developers)!)

**Comment by nedrichards** [comment]
"no, teams are private to an org" (I can't even see that URL for example)

File: org.worldpossible.ScriptLauncher.yml (hunk 4 above)
Comment (barthalion) [reply org.worldpossible.ScriptLauncher.yml:9]: Nick said so under some review, I haven't checked it myself.

**Pull request closed without merging**